src/controller.coffee

```diff
@@ -4287,15 +4316,31 @@ Editor::addLineNumberForLine = (line) ->
     @lineNumberTags[line] = lineDiv
 
   lineDiv.className = 'droplet-gutter-line'
-  if @gutterDecorations[line]
-    lineDiv.className += ' ' + @gutterDecorations[line].join(' ')
-  lineDiv.style.top = "#{treeView.bounds[line].y - @scrollOffsets.main.y}px"
+
+  # Add annotation mouseover text
+  # and graphics
+  if @annotations[line]?
+    lineDiv.className += ' droplet_' + getMostSevereAnnotationType(@annotations[line])
+    lineDiv.style.backgroundPosition = "2px #{treeView.distanceToBase[line].above - @view.opts.textHeight - @fontAscent}px"
+    lineDiv.title = @annotations[line].map((x) -> x.text).join('\n')
+
+  # Add breakpoint graphics
+  if @breakpoints[line]
+    lineDiv.className += ' droplet_breakpoint'
+
+  lineDiv.style.top = "#{treeView.bounds[line].y}px"
   lineDiv.style.paddingTop = "#{treeView.distanceToBase[line].above - @view.opts.textHeight - @fontAscent}px"
   lineDiv.style.height =  treeView.bounds[line].height + 'px'
   lineDiv.style.fontSize = @fontSize + 'px'
 
   @lineNumberWrapper.appendChild lineDiv
 
+getMostSevereAnnotationType = (arr) ->
+  for el, i in arr
+    if el.type is 'error'
+      return 'error'
+  return 'warning'
```

Comment:
Are annotations guaranteed to always be 'error' or 'warning'? (I'm not familiar with the spec)